src/box/replication.cc

```diff
@@ -319,6 +333,17 @@ replica_delete(struct replica *replica)
 	free(replica);
 }
 
+static void
+replica_free(struct replica *replica)
```

Comment:
We have `replica_delete` already. Can't you reuse it? It's ugly to have `replica_free` and `replica_delete` both of which free a replica object.

Reply:
Fixed.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/gc.c b/src/box/gc.c
index 7ab2df9f31..e7fc201d86 100644
--- a/src/box/gc.c
+++ b/src/box/gc.c
@@ -396,6 +396,8 @@ gc_delay_unref(void)
 static void
 gc_schedule_cleanup(void)
 {
+	if (gc.cleanup_fiber == NULL)
+		return;
 	/*
 	 * Do not wake up the background fiber if it's executing
 	 * the garbage collection procedure right now, because
diff --git a/src/box/replication.cc b/src/box/replication.cc
index 2be69c150f..e50fc0c637 100644
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -80,9 +80,9 @@ enum bootstrap_strategy bootstrap_strategy = BOOTSTRAP_STRATEGY_INVALID;
 
 enum replicaset_state replicaset_state = REPLICASET_BOOTSTRAP;
 
-/* Free replica resources. */
+/** Free replica resources. */
 static void
-replica_free(struct replica *replica);
+replica_delete(struct replica *replica);
 
 static int
 replica_compare_by_uuid(const struct replica *a, const struct replica *b)
@@ -248,11 +248,11 @@ replication_free(void)
 	while (!rlist_empty(&replicaset.anon)) {
 		replica = rlist_shift_entry(&replicaset.anon,
 					    typeof(*replica), in_anon);
-		replica_free(replica);
+		replica_delete(replica);
 	}
 	replica_hash_foreach_safe(&replicaset.hash, replica, next) {
 		replica_hash_remove(&replicaset.hash, replica);
-		replica_free(replica);
+		replica_delete(replica);
 	}
 	diag_destroy(&replicaset.applier.diag);
 	trigger_destroy(&replicaset.on_ack);
@@ -323,23 +323,13 @@ replica_new(void)
 
 static void
 replica_delete(struct replica *replica)
-{
-	assert(replica_is_orphan(replica));
-	if (replica->relay != NULL)
-		relay_delete(replica->relay);
-	if (replica->gc != NULL)
-		gc_consumer_unregister(replica->gc);
-	TRASH(replica);
-	free(replica);
-}
-
-static void
-replica_free(struct replica *replica)
 {
 	if (replica->relay != NULL)
 		relay_delete(replica->relay);
 	if (replica->applier != NULL)
 		applier_delete(replica->applier);
+	if (replica->gc != NULL)
+		gc_consumer_unregister(replica->gc);
 	TRASH(replica);
 	free(replica);
 }
```

</summary>